lib/nm.ts

```diff
@@ -0,0 +1,218 @@
+import * as dbus from 'dbus-as-promised';
+import * as os from 'os';
+import { Options } from './index';
+
+type ActiveConnection = string;
+type Connection = ActiveConnection;
+
+class NetworkManager {
+	private wiredReference:
+		| {
+				activeConn: ActiveConnection;
+				conn: Connection;
+		  }
+		| undefined;
+	private wirelessReference:
+		| {
+				activeConn: ActiveConnection;
+				conn: Connection;
+		  }
+		| undefined;
+
+	constructor(
+		private options: Options['network'],
+		private bus = dbus.getBus('system'),
+	) {
+		// Cleanup code
+		process.on('SIGINT', this.disconnect);
+	}
+
+	private static stringToArrayOfBytes(str: string): Array<number> {
+		let bytes = [];
+		for (var i = 0; i < str.length; ++i) {
+			bytes.push(str.charCodeAt(i));
+		}
+		return bytes;
+	}
+
+	private static generateId(): string {
+		return Math.random()
+			.toString(36)
+			.substring(2, 10);
+	}
+
+	private static wiredTemplate(method: string): any {
+		return {
+			connection: {
+				id: NetworkManager.generateId(),
+				type: '802-3-ethernet',
+				autoconnect: false,
+			},
+			ipv4: { method },
+			ipv6: { method: 'ignore' },
+		};
+	}
+
+	private static wirelessTemplate(
+		method: string,
+		ssid: string,
+		psk: string,
+	): any {
```

Comment:
```suggestion
	) {
```

Let typescript infer the type here rather than returning any (which can potentially cause issues in any function calling this). (Or strongly type the return)